File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-5.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int left { 0 };
        int total { 0 };
        int res { INT_MAX };

        for (std::size_t right { 0 }; right < nums.size(); right++) {
            total += nums[right];

            while (total >= target) {
                res = (right - left + 1) < res ? right - left + 1 : res;
                total -= nums[left];
                left++;
            }
        }

        return res == INT_MAX ? 0 : res;
    }
};

File: Data Structures & Algorithms/search-insert-position/submission-5.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int res = nums.length;
        int left = 0;
        int right = nums.length - 1;

        while (left <= right) {
            int mid = left + (right - left) / 2;
            if (nums[mid] == target) {
                return mid;
            }

            if (nums[mid] > target) {
                res = mid;
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }

        return res;
    }
}